### ISSUE TITLE
Week 5 Ready For Review

### ISSUE THREAD

**Issue by ctkrause** [ctkrause/IndieProjectCalebKrause#2]
@pawaitemadisoncollege - Week 5 is complete and ready for review. 

What were your key learning points/takeaways?
- Follow the recipe. All the pieces are there; just follow the recipe and you'll be annotating beautiful one to many relationships in no time. Take your time. Make sure you've updated your configs, added the table to both dbs, and annotated everything correctly.

What challenged you?
- Figuring out how to test everything was my only challenge. Obviously, it starts to get a little bit complex when you've got multiple tables that are related by a foreign key and you have to know how to target what you're looking to compare exactly. IntelliJ tried to help, but it was easier to just think through things logically.

What problems did you solve and what resources did you use to solve them?
- Hmm, I didn't really have any issues. Syntactical errors were my only issue and those were fairly easy to resolve.

**Comment by pawaitemadisoncollege** [comment]
You have some really good momentum on your indie project, @ctkrause!  Sounds like you did a very good job with all the details here and have a good process for adding more entities/tables.

- [x] This is a good time to replace the entity-specific daos with the abstract/generic dao.  Remember that you'll still need to have specific unit tests for each entity-dao to ensure the annotations and tables are set up correctly, it's just that the unit tests will utilize that generic Dao instead. 
- [x] The delete unit tests would be more thorough by verifying the other side of the relationship after the delete. Example: if a user is deleted, should their role also be deleted? Is it? What about the role? If a role is removed, does the user still stick around?

**Comment by ctkrause** [comment]
Hadn't watched the follow up videos when I was doing the one-to-many addition, but now that I have my comparison within my insert test of newUser to insertedUser  is behaving correctly. I've also added the generic dao and removed the User and Role specific daos I had. 
Additionally, I added testing in my deleteSuccess() within RoleDaoTest to ensure the existing user was not removed when I deleted a row from the role table.
